h2o-core/src/test/java/water/fvec/FrameTest.java

```diff
@@ -3,21 +3,50 @@
 import org.junit.BeforeClass;
 import org.junit.Ignore;
 import org.junit.Test;
+import water.H2O;
 import water.Key;
 import water.Scope;
 import water.TestUtil;
+import static org.junit.Assert.*;
 
 import static org.junit.Assert.assertTrue;
 
 /**
- * Created by tomas on 3/16/16.
+ * Tests for Frame.java
  */
 public class FrameTest extends TestUtil {
   @BeforeClass
   public static void setup() {
     stall_till_cloudsize(1);
   }
 
+  @Test
+  public void testRemoveColumn() {
+    Scope.enter();
+    Frame testData = parse_test_file(Key.make("test_deep_select_1"), "smalldata/sparse/created_frame_binomial.svm.zip");
+    try {
+      // dataset to split
+      int initialSize = testData.numCols();
+      testData.remove(-1);
+      assertEquals(initialSize, testData.numCols());
+      testData.remove(0);
+      assertEquals(initialSize - 1, testData.numCols());
+      assertEquals("C2", testData._names[0]);
+      testData.remove(initialSize - 2);
+      assertEquals(initialSize - 2, testData.numCols());
+      assertEquals("C" + (initialSize - 1), testData._names[initialSize - 3]);
+      testData.remove(42);
+      assertEquals(initialSize - 3, testData.numCols());
+      assertEquals("C43", testData._names[41]);
+      assertEquals("C45", testData._names[42]);
+    } finally {
+      Scope.exit();
+      testData.delete();
+      H2O.STORE.clear(); // TODO(vlad): figure out how come it's dirty after deletion of columns
```

Comment:
This is bad since it can hide memory leak. You should delete data you created manually.

Delete each removed column `testData.remove(-1).remove()`

At the end, you are correctly removing the frame: `testData.delete()`